src/main/java/org/spburegistry/backend/controller/TagController.java

```diff
@@ -0,0 +1,40 @@
+package org.spburegistry.backend.controller;
+
+import org.spburegistry.backend.dto.TagTO;
+import org.spburegistry.backend.service.TagService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+
+@RequestMapping("/data/tags")
+@RestController
+public class TagController {
+
+    @Autowired
+    private TagService tagService;
+    
+    @GetMapping
+    public Iterable<TagTO> getAllTags() {
+        return tagService.findAll();
+    }
+
+    @GetMapping("/tag/id")
+    public TagTO getTagById(@RequestParam Long id) {
+        return tagService.findTagById(id);
+    }
+
+    @GetMapping("/tag/regex")
```

Comment:
I think it is better to name route like substring because it reflects what will you do on db better. You won't fins tags by regular expressions

Reply:
Done